ovs/datapath.go

```diff
@@ -0,0 +1,342 @@
+// for the ovs DataPaths
+type DataPathReader interface {
+	// Version is the method used to get the version of ovs-dpctl
+	Version() (string, error)
+	// GetDataPath is the method that returns all DataPaths setup
+	// for an ovs switch
+	GetDataPath() ([]string, error)
+}
+
+// DataPathWriter is the interface defining the wrtie operations
+// for the ovs DataPaths
+type DataPathWriter interface {
+	// AddDataPath is the method used to add a datapath to the switch
+	AddDataPath(string) ([]byte, error)
+	// DelDataPath is the method used to remove a datapath from the switch
+	DelDataPath(string) ([]string, error)
+}
+
+// ConnTrackReader is the interface defining the read operations
+// of ovs conntrack
+type ConnTrackReader interface {
+	// GetCTLimits is the method used to querying conntrack limits for a
+	// datapath on a switch
+	GetCTLimits(string, []uint64) (ConnTrackOutput, error)
+}
+
+// ConnTrackWriter is the interface defining the write operations
+// of ovs conntrack
+type ConnTrackWriter interface {
+	// SetCTLimits is the method used to setup a limit for an ofport (zone)
+	// belonging to a datapath of  a switch
+	SetCTLimits(string) (string, error)
+	// DelCTLimits is the method used to remove a limit to an ofport (zone)
+	// belonging to a datapath of a switch
+	DelCTLimits(string, []uint64) (string, error)
+}
+
+// CLI is an interface defining a contract for executing a command.
+// Implementation of shell cli is done by the Client concrete type
+type CLI interface {
+	Exec(args ...string) ([]byte, error)
+}
+
+// DataPathService defines the concrete type used for DataPath operations
+// supported by the ovs-dpctl command
+type DataPathService struct {
+	// We define here a CLI interface making easier to mock ovs-dpctl command
+	// as in github.com/digitalocean/go-openvswitch/ovs/datapath_test.go
+	CLI
+}
+
+// NewDataPathService is a builder for the DataPathService.
+// sudo is defined as a default option.
+func NewDataPathService() *DataPathService {
+	return &DataPathService{
+		CLI: &OvsCLI{
+			c: New(Sudo()),
+		},
+	}
+}
+
+// Version retruns the ovs-dptcl --version currently installed
+func (dp *DataPathService) Version() (string, error) {
+	args := []string{"--version"}
+	result, err := dp.CLI.Exec(args...)
+	if err != nil {
+		return "", err
+	}
+
+	return string(result), nil
+}
+
+// GetDataPaths returns the output of the command 'ovs-dpctl dump-dps'
+func (dp *DataPathService) GetDataPaths() ([]string, error) {
+	args := []string{"dump-dps"}
+	result, err := dp.CLI.Exec(args...)
+	if err != nil {
+		return nil, err
+	}
+
+	return strings.Split(string(result), "\n"), nil
+}
+
+// AddDataPath create a Datapath with the command 'ovs-dpctl add-dp <DP>'
+// It takes one argument, the required DataPath Name and returns an error
+// if it failed
+func (dp *DataPathService) AddDataPath(dpName string) ([]byte, error) {
+	args := []string{"add-dp", dpName}
+
+	return dp.CLI.Exec(args...)
```

Comment:
The variadic argument thing should also work where there are multiple arguments. This could be written like this, I think.

```
return dp.CLI.Exec("add-dp", dpName)
```